File: e_commerce_website/test4/controllers/postController.js
// Import the Post model to interact with the database
const Post = require('../models/Post');

// Function to get all posts from the database
const getAllPosts = async (req, res) => {
    try {
        // Retrieve all posts using the Post model
        const posts = await Post.find();
        // Send the retrieved posts as a JSON response
        res.json(posts);
    } catch (error) {
        // Handle any errors by sending a 500 status and error message
        res.status(500).json({ message: 'Server Error', error: error.message });
    }
};

// Function to create a new post in the database
const createPost = async (req, res) => {
    try {
        // Extract title and content from the request body
        const { title, content } = req.body;
        
        // Create a new Post instance with the provided data
        const newPost = new Post({
            title,
            content,
            date: new Date() // Set the current date
        });

        // Save the new post to the database
        await newPost.save();

        // Send a success response with the created post
        res.status(201).json(newPost);
    } catch (error) {
        // Handle any errors by sending a 400 status and error message
        res.status(400).json({ message: 'Error creating post', error: error.message });
    }
};

// Export the controller functions for use in routes
module.exports = {
    getAllPosts,
    createPost
};

File: e_commerce_website/test2/src/controllers/post_controller.py
from flask import request, jsonify
from src.models.posts import get_all_posts, get_post_by_id, create_post

def handle_get_posts():
    """
    Handles the request to get all posts.
    """
    try:
        posts = get_all_posts()
        return jsonify(posts), 200
    except Exception as e:
        return jsonify({"error": str(e)}), 500

def handle_get_post(post_id):
    """
    Handles the request to get a specific post by ID.
    """
    try:
        post = get_post_by_id(post_id)
        if post:
            return jsonify(post), 200
        else:
            return jsonify({"error": "Post not found"}), 404
    except Exception as e:
        return jsonify({"error": str(e)}), 500

def handle_create_post():
    """
    Handles the request to create a new post.
    """
    try:
        data = request.get_json()
        title = data.get('title')
        content = data.get('content')
        author = data.get('author')

        if not title or not content or not author:
            return jsonify({"error": "Missing required fields"}), 400

        new_post = create_post(title, content, author)
        return jsonify(new_post), 201
    except Exception as e:
        return jsonify({"error": str(e)}), 500